praxis/optimizers.py

```diff
@@ -2767,16 +2754,38 @@ def _run_base_tx():
               jax.tree_map(lambda u: jnp.zeros_like(u, dtype=jnp.float32),
                            updates), emission_base_state)
 
-    # PAX makes use of vectorized map for repeated layers. XLA currently doesn't
-    # handle conds with-in vmap well and thus calls into both branches with a
-    # select. Here we rewrite a lax.cond as while_loop to get around this issue
-    # and get faster step time.
-    new_updates, new_accumulated_update, new_base_state = while_cond(
-        should_emit, _run_base_tx, [
-            jax.tree_map(jnp.zeros_like, updates), new_accumulated_update,
-            state.base_state  # pytype: disable=attribute-error  # jax-ndarray
-        ])
+    if accumulation_use_cond_op:
+        def _continue_accumulating():
+          return (jax.tree_map(jnp.zeros_like, updates), new_accumulated_update,
+                  state.base_state) # pytype: disable=attribute-error # jax-ndarray
 
+        new_updates, new_accumulated_update, new_base_state = lax.cond(
+            should_emit, _run_base_tx, _continue_accumulating)
+    else:
+        # PAX makes use of vectorized map for repeated layers. XLA currently doesn't
+        # handle conds with-in vmap well and thus calls into both branches with a
+        # select. Here we rewrite a lax.cond as while_loop to get around this issue
```

Comment:
Is there an open bug with XLA? If not could one be filed on https://github.com/openxla/xla/issues ?

Reply:
I don't know the case, this is the original comments that describes the problem, but I don't see it in PAXML model.